hugr-core/src/export.rs

```diff
@@ -0,0 +1,726 @@
+                model::Operation::DeclareAlias { decl }
+            }
+
+            OpType::AliasDefn(alias) => {
+                let value = self.export_type(&alias.definition);
+                // TODO: We should support aliases with different types and with parameters
+                let r#type = self.module.insert_term(model::Term::Type);
+                let decl = self.bump.alloc(model::AliasDecl {
+                    name: &alias.name,
+                    params: &[],
+                    r#type,
+                });
+                model::Operation::DefineAlias { decl, value }
+            }
+
+            OpType::Call(call) => {
+                // TODO: If the node is not connected to a function, we should do better than panic.
+                let node = self.connected_function(node).unwrap();
+                let name = model::GlobalRef::Named(self.get_func_name(node).unwrap());
+
+                let mut args = BumpVec::new_in(self.bump);
+                args.extend(call.type_args.iter().map(|arg| self.export_type_arg(arg)));
+                let args = args.into_bump_slice();
+
+                let func = self
+                    .module
+                    .insert_term(model::Term::ApplyFull { global: name, args });
+                model::Operation::CallFunc { func }
+            }
+
+            OpType::LoadFunction(load) => {
+                // TODO: If the node is not connected to a function, we should do better than panic.
+                let node = self.connected_function(node).unwrap();
+                let name = model::GlobalRef::Named(self.get_func_name(node).unwrap());
+
+                let mut args = BumpVec::new_in(self.bump);
+                args.extend(load.type_args.iter().map(|arg| self.export_type_arg(arg)));
+                let args = args.into_bump_slice();
+
+                let func = self
+                    .module
+                    .insert_term(model::Term::ApplyFull { global: name, args });
+                model::Operation::LoadFunc { func }
+            }
+
+            OpType::Const(_) => todo!("Export const nodes?"),
+            OpType::LoadConstant(_) => todo!("Export load constant?"),
+
+            OpType::CallIndirect(_) => make_custom(OP_FUNC_CALL_INDIRECT),
+
+            OpType::Tag(tag) => model::Operation::Tag { tag: tag.tag as _ },
+
+            OpType::TailLoop(_) => {
+                regions = self.bump.alloc_slice_copy(&[self.export_dfg(node)]);
+                model::Operation::TailLoop
+            }
+
+            OpType::Conditional(_) => {
+                regions = self.export_conditional_regions(node);
+                model::Operation::Conditional
+            }
+
+            // Opaque/extension operations should in the future support having multiple different
+            // regions of potentially different kinds. At the moment, we check if the node has any
+            // children, in which case we create a dataflow region with those children.
```

Comment:
// We assume that the regions will always have dataflow kind.

Reply:
For now